pandas/core/tools/datetimes.py

```diff
@@ -698,7 +708,8 @@ def coerce(values):
             except (TypeError, ValueError) as e:
                 raise ValueError("cannot assemble the datetimes [{value}]: "
                                  "{error}".format(value=value, error=e))
-
+    if not box:
```

Comment:
does box=False actually make sense for this?

Reply:
Although probably uncommon, I think it's reasonable to return a numpy array from a unit mapping.